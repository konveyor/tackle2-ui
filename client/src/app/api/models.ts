export interface PageQuery {
  page: number;
  perPage: number;
}

export interface SortByQuery {
  index: number;
  direction: "asc" | "desc";
}

export interface PageRepresentation<T> {
  meta: Meta;
  data: T[];
}

export interface Meta {
  count: number;
}

// Controls

export interface BusinessService {
  id?: number;
  name: string;
  description?: string;
  owner?: Stakeholder;
}

export interface Stakeholder {
  id?: number;
  name: string;
  email: string;
  jobFunction?: JobFunction;
  stakeholderGroups?: Ref[];
  businessServices?: Ref[];
}

export interface StakeholderGroup {
  id?: number;
  name: string;
  description?: string;
  stakeholders?: Ref[];
}
export interface Ref {
  id?: number;
  name: string;
}

export interface JobFunction {
  id?: number;
  name: string;
  stakeholders?: Array<Ref>;
}

export interface TagType {
  id?: number;
  name: string;
  rank?: number;
  colour?: string;
  tags?: Tag[];
}

export interface Tag {
  id?: number;
  name: string;
  tagType?: Ref;
}

// Application inventory
export type ProposedAction =
  | "rehost"
  | "replatform"
  | "refactor"
  | "repurchase"
  | "retire"
  | "retain";

export type EffortEstimate = "small" | "medium" | "large" | "extra_large";

export type ImportSummaryStatus = "Completed" | "In Progress" | "Failed";

export interface Repository {
  kind?: string;
  branch?: string;
  path?: string;
  url?: string;
}

export interface Application {
  id?: number;
  name: string;
  description?: string;
  comments?: string;
  businessService?: Ref;
  tags?: Ref[];
  review?: Ref;
  identities?: Ref[];
  repository?: Repository;
  binary?: string;
}

export interface Review {
  id?: number;
  proposedAction: ProposedAction;
  effortEstimate: EffortEstimate;
  businessCriticality: number;
  workPriority: number;
  comments?: string;
  application?: Application;
}

export interface ApplicationDependency {
  id?: number;
  from: Application;
  to: Application;
}

export interface ApplicationAdoptionPlan {
  applicationId: number;
  applicationName: string;
  positionX: number;
  positionY: number;
  effort: number;
  decision: ProposedAction;
  effortEstimate: string;
}

export interface ApplicationImportSummary {
  id: number;
  filename: string;
  createUser: string;
  importTime: string;
  validCount: number;
  invalidCount: number;
  importStatus: ImportSummaryStatus;
}

export interface ApplicationImport {
  "Application Name": string;
  errorMessage: string;
  isValid: boolean;
}

export interface BulkCopyReview {
  id?: number;
  sourceReview: number;
  targetApplications: number[];
  completed?: boolean;
}

export type IdentityKind = "source" | "maven" | "proxy";

export interface Identity {
  id?: number;
  name?: string;
  description?: string;
  kind?: IdentityKind;
  createUser?: string;
  encrypted?: string;
  key?: string;
  keyFilename?: string;
  password?: string;
  user?: string;
  updateUser?: string;
  settings?: string;
  settingsFilename?: string;
}

export interface Proxy {
  host: string;
  kind: "http" | "https";
  port: number;
  excluded: Array<string>;
  identity?: Ref;
  createTime?: string;
  createUser?: string;
  id: any;
  enabled: boolean;
}

// Pathfinder

export type AssessmentStatus = "EMPTY" | "STARTED" | "COMPLETE";
export type Risk = "GREEN" | "AMBER" | "RED" | "UNKNOWN";

export interface Assessment {
  id?: number;
  applicationId: number;
  status: AssessmentStatus;
  stakeholders?: number[];
  stakeholderGroups?: number[];
  questionnaire: Questionnaire;
}

export interface Questionnaire {
  categories: QuestionnaireCategory[];
}

export interface QuestionnaireCategory {
  id: number;
  order: number;
  title?: string;
  comment?: string;
  questions: Question[];
}

export interface Question {
  id: number;
  order: number;
  question: string;
  description: string;
  options: QuestionOption[];
}

export interface QuestionOption {
  id: number;
  order: number;
  option: string;
  checked: boolean;
  risk: Risk;
}

export interface AssessmentRisk {
  assessmentId: number;
  applicationId: number;
  risk: Risk;
}

export interface AssessmentQuestionRisk {
  category: string;
  question: string;
  answer: string;
  applications: number[];
}

export interface AssessmentConfidence {
  assessmentId: number;
  applicationId: number;
  confidence: number;
}

export interface BulkCopyAssessment {
  bulkId?: number;
  fromAssessmentId: number;
  applications: { applicationId: number }[];
  completed?: boolean;
}

// Pagination

export interface BusinessServicePage {
  _embedded: {
    "business-service": BusinessService[];
  };
  total_count: number;
}

export interface StakeholderPage {
  _embedded: {
    stakeholder: Stakeholder[];
  };
  total_count: number;
}

export interface StakeholderGroupPage {
  _embedded: {
    "stakeholder-group": StakeholderGroup[];
  };
  total_count: number;
}

export type JobFunctionPage = Array<JobFunction>;

export interface TagTypePage {
  _embedded: {
    "tag-type": TagType[];
  };
  total_count: number;
}

export interface TagTypePage {
  _embedded: {
    "tag-type": TagType[];
  };
  total_count: number;
}

export interface ApplicationPage {
  _embedded: {
    application: Application[];
  };
  total_count: number;
}

export interface ApplicationDependencyPage {
  _embedded: {
    "applications-dependency": ApplicationDependency[];
  };
  total_count: number;
}

export interface ApplicationImportSummaryPage {
  _embedded: {
    "import-summary": ApplicationImportSummary[];
  };
  total_count: number;
}

export interface ApplicationImportPage {
  _embedded: {
    "application-import": ApplicationImport[];
  };
  total_count: number;
}

export type SettingTypes = {
  "download.csv.enabled": boolean;
  "download.html.enabled": boolean;
  "git.insecure.enabled": boolean;
  "mvn.dependencies.update.forced": boolean;
  "mvn.insecure.enabled": boolean;
  "review.assessment.required": boolean;
  "svn.insecure.enabled": boolean;
  "ui.bundle.order": number[];
};

export type Setting<K extends keyof SettingTypes> = {
  key: K;
  value: SettingTypes[K];
};

// Analysis Task

export type TaskState =
  | "not supported"
  | "Canceled"
  | "Created"
  | "Succeeded"
  | "Failed"
  | "Running"
  | "No task"
  | "Ready"
  | "Pending"
  | "Postponed";

export interface Task {
  id?: number;
  createTime?: string;
  application: { id: number };
  name: string;
  addon: string;
  data: TaskData;
  error?: string;
  image?: string;
  started?: string;
  terminated?: string;
  state?: TaskState;
  job?: string;
  report?: TaskReport;
}

export interface TaskData {
  output: string;
  mode: {
    binary: boolean;
    withDeps: boolean;
    artifact: string;
    diva: boolean;
  };
  targets: string[];
  sources: string[];
  scope: {
    withKnown: boolean;
    packages: {
      included: string[];
      excluded: string[];
    };
  };
  rules?: {
    path: string;
    tags: {
      excluded: string[];
    };
    bundles: Ref[];
  };
}

interface TaskReport {
  activity: string[];
  completed: number;
  createTime: string;
  createUser: string;
  error: string;
  id: number;
  status: string;
  task: number;
  total: number;
  updateUser: string;
}

export interface TaskgroupTask {
  name: string;
  data: any;
  application: Ref;
}

export interface Taskgroup {
  id?: number;
  name: string;
  addon: string;
  data: TaskData;
  tasks: TaskgroupTask[];
}

export interface Cache {
  path: string;
  capacity: string;
  used: string;
  exists: boolean;
}

export interface ITypeOptions {
  key: string;
  value: string;
}

export interface RuleBundleImage {
  id: number;
  name?: string;
}

export enum RuleBundleKind {
  CATEGORY = "category",
}

export interface RuleBundle {
  id: number;
  name: string;
  description: string;
  image: RuleBundleImage;
  kind?: RuleBundleKind;
  rulesets: Ruleset[];
  custom: boolean;
  repository?: Repository;
}
export interface Metadata {
  target: string;
  source?: string;
}
export interface Ruleset {
  name: string;
  metadata: Metadata;
  file?: {
    id: number;
  };
}

export interface TableRule {
  name: string;
  source: string | null;
  target: string | null;
  total: number;
  fileID?: number;
}

export interface IReadFile {
  fileName: string;
  fullFile: File;
  loadError?: DOMException;
  loadPercentage?: number;
  loadResult?: "danger" | "success";
  data?: string;
  responseID?: number;
}
