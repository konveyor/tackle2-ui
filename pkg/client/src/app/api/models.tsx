import { boolean } from "yup";

export interface PageQuery {
  page: number;
  perPage: number;
}

export interface SortByQuery {
  index: number;
  direction: "asc" | "desc";
}

export interface PageRepresentation<T> {
  meta: Meta;
  data: T[];
}

export interface Meta {
  count: number;
}

// Controls

export interface BusinessService {
  id?: number;
  name: string;
  description?: string;
  owner?: Stakeholder;
}

export interface Stakeholder {
  id?: number;
  displayName: string;
  email: string;
  jobFunction?: JobFunction;
  stakeholderGroups?: StakeholderGroup[];
}

export interface StakeholderGroup {
  id?: number;
  name: string;
  description?: string;
  stakeholders?: Stakeholder[];
}

export interface JobFunction {
  id?: number;
  role: string;
}

export interface TagType {
  id?: number;
  name: string;
  rank?: number;
  colour?: string;
  tags?: Tag[];
}

export interface Tag {
  id?: number;
  name: string;
  tagType?: TagType;
}

// Application inventory
export type ProposedAction =
  | "rehost"
  | "replatform"
  | "refactor"
  | "repurchase"
  | "retire"
  | "retain";

export type EffortEstimate = "small" | "medium" | "large" | "extra_large";

export type ImportSummaryStatus = "Completed" | "In Progress" | "Failed";

export interface Application {
  id?: number;
  name: string;
  description?: string;
  comments?: string;
  businessService?: string;
  tags?: string[];
  review?: Review;
}

export interface Review {
  id?: number;
  proposedAction: ProposedAction;
  effortEstimate: EffortEstimate;
  businessCriticality: number;
  workPriority: number;
  comments?: string;
  application?: Application;
}

export interface ApplicationDependency {
  id?: number;
  from: Application;
  to: Application;
}

export interface ApplicationAdoptionPlan {
  applicationId: number;
  applicationName: string;
  positionX: number;
  positionY: number;
  effort: number;
  decision: ProposedAction;
  effortEstimate: string;
}

export interface ApplicationImportSummary {
  id: number;
  filename: string;
  createUser: string;
  importTime: string;
  validCount: number;
  invalidCount: number;
  importStatus: ImportSummaryStatus;
}

export interface ApplicationImport {
  "Application Name": string;
  errorMessage: string;
  isValid: boolean;
}

export interface BulkCopyReview {
  id?: number;
  sourceReview: number;
  targetApplications: number[];
  completed?: boolean;
}
export interface Identity {
  id?: number;
  name: string;
  description?: string;
  kind: string;
  createUser: string;
}

// Pathfinder

export type AssessmentStatus = "EMPTY" | "STARTED" | "COMPLETE";
export type Risk = "GREEN" | "AMBER" | "RED" | "UNKNOWN";

export interface Assessment {
  id?: number;
  applicationId: number;
  status: AssessmentStatus;
  stakeholders?: number[];
  stakeholderGroups?: number[];
  questionnaire: Questionnaire;
}

export interface Questionnaire {
  categories: QuestionnaireCategory[];
}

export interface QuestionnaireCategory {
  id: number;
  order: number;
  title?: string;
  comment?: string;
  questions: Question[];
}

export interface Question {
  id: number;
  order: number;
  question: string;
  description: string;
  options: QuestionOption[];
}

export interface QuestionOption {
  id: number;
  order: number;
  option: string;
  checked: boolean;
  risk: Risk;
}

export interface AssessmentRisk {
  assessmentId: number;
  applicationId: number;
  risk: Risk;
}

export interface AssessmentQuestionRisk {
  category: string;
  question: string;
  answer: string;
  applications: number[];
}

export interface AssessmentConfidence {
  assessmentId: number;
  applicationId: number;
  confidence: number;
}

export interface BulkCopyAssessment {
  bulkId?: number;
  fromAssessmentId: number;
  applications: { applicationId: number }[];
  completed?: boolean;
}

// Pagination

export interface BusinessServicePage {
  _embedded: {
    "business-service": BusinessService[];
  };
  total_count: number;
}

export interface StakeholderPage {
  _embedded: {
    stakeholder: Stakeholder[];
  };
  total_count: number;
}

export interface StakeholderGroupPage {
  _embedded: {
    "stakeholder-group": StakeholderGroup[];
  };
  total_count: number;
}

export interface JobFunctionPage {
  _embedded: {
    "job-function": JobFunction[];
  };
  total_count: number;
}

export interface TagTypePage {
  _embedded: {
    "tag-type": TagType[];
  };
  total_count: number;
}

export interface TagTypePage {
  _embedded: {
    "tag-type": TagType[];
  };
  total_count: number;
}

export interface ApplicationPage {
  _embedded: {
    application: Application[];
  };
  total_count: number;
}

export interface ApplicationDependencyPage {
  _embedded: {
    "applications-dependency": ApplicationDependency[];
  };
  total_count: number;
}

export interface ApplicationImportSummaryPage {
  _embedded: {
    "import-summary": ApplicationImportSummary[];
  };
  total_count: number;
}

export interface ApplicationImportPage {
  _embedded: {
    "application-import": ApplicationImport[];
  };
  total_count: number;
}
export interface IdentityPage {
  _embedded: {
    identity: Identity[];
  };
  total_count: number;
}
