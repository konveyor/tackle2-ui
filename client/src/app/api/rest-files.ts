import axios from "axios";

import { HUB, HEADERS } from "./rest";
import { HubFile, IReadFile } from "./models";

export const FILES = HUB + "/files";

export const createFile = ({
  formData,
  file,
}: {
  formData: FormData;
  file: IReadFile;
}) =>
  axios
    .post<HubFile>(`${FILES}/${file.fileName}`, formData, {
      headers: HEADERS.file,
    })
    .then((response) => response.data);

export const getTextFileById = (id: number) =>
  axios
    .get<string>(`${FILES}/${id}`, { headers: HEADERS.plain })
    .then((response) => response.data);
