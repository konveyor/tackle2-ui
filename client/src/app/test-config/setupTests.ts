import "@testing-library/jest-dom";
import { server } from "@mocks/server";

const mockInitialized = false;

jest.mock("@react-keycloak/web", () => {
  const originalModule = jest.requireActual("@react-keycloak/web");
  return {
    ...originalModule,
    useKeycloak: () => [mockInitialized],
  };
});

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "localhost:3000/example/path",
  }),
}));

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

// Uncomment the following to see request logging in the console:

// server.events.on("request:start", (req) => {
//   console.log(`Handling a request to ${req.url.href}`);
// });

// server.events.on("request:match", (req) => {
//   console.log(`Request to ${req.url.href} was matched with a handler`);
// });

// server.events.on("request:unhandled", (req) => {
//   console.warn(`Request to ${req.url.href} was not handled`);
// });
