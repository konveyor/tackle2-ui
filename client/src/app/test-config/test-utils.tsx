import React, { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { Application, Archetype, Assessment } from "@app/api/models";
import { RenderHookOptions, renderHook } from "@testing-library/react-hooks";

import { createContext, useContext } from "react";

const QueryClientContext = createContext<QueryClient | undefined>(undefined);

const AllTheProviders: FC<{
  children: React.ReactNode;
  queryClient?: QueryClient;
}> = ({ children, queryClient }) => {
  const internalQueryClient =
    queryClient ||
    new QueryClient({
      defaultOptions: {
        queries: {
          retry: false,
          cacheTime: 1000,
        },
      },
    });
  return (
    <QueryClientProvider client={internalQueryClient}>
      {children}
    </QueryClientProvider>
  );
};

export const useQueryClientContext = () => {
  const context = useContext(QueryClientContext);
  if (context === undefined) {
    throw new Error(
      "useQueryClientContext must be used within a QueryClientContext.Provider"
    );
  }
  return context;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

const customRenderHook = <TProps, TResult>(
  callback: (props: TProps) => TResult,
  options?: Omit<RenderHookOptions<TProps>, "wrapper"> & {
    queryClient?: QueryClient;
  }
) => {
  const { queryClient, ...rest } = options || {};
  const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => (
    <AllTheProviders queryClient={queryClient}>{children}</AllTheProviders>
  );

  return renderHook(callback, { wrapper: Wrapper as React.FC, ...rest });
};

export * from "@testing-library/react";

export { customRender as render };
export { customRenderHook as renderHook };

export const createMockAssessment = (
  overrides: Partial<Assessment> = {}
): Assessment => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    required: true,
    ...overrides,
  } as Assessment;
};

export const createMockApplication = (overrides: Partial<Application> = {}) => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    ...overrides,
  } as Application;
};

export const createMockArchetype = (overrides: Partial<Archetype> = {}) => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    ...overrides,
  } as Archetype;
};
