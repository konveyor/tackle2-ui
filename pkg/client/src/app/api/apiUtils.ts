import { AxiosPromise } from "axios";
import {
  Application,
  ApplicationDependency,
  ApplicationDependencyPage,
  ApplicationImport,
  ApplicationImportPage,
  ApplicationImportSummary,
  ApplicationImportSummaryPage,
  ApplicationPage,
  BusinessService,
  BusinessServicePage,
  Identity,
  IdentityPage,
  JobFunction,
  JobFunctionPage,
  PageRepresentation,
  Stakeholder,
  StakeholderGroup,
  StakeholderGroupPage,
  StakeholderPage,
  TagType,
  TagTypePage,
} from "./models";
import {
  ApplicationSortBy,
  BusinessServiceSortBy,
  getApplicationDependencies,
  getApplications,
  getBusinessServices,
  getJobFunctions,
  getStakeholderGroups,
  getStakeholders,
  getTagTypes,
  JobFunctionSortBy,
  StakeholderGroupSortBy,
  StakeholderSortBy,
  TagTypeSortBy,
  TagTypeSortByQuery,
} from "./rest";

export const getAllBusinessServices = () => {
  return getBusinessServices(
    {},
    { page: 1, perPage: 1000 },
    { field: BusinessServiceSortBy.NAME }
  );
};

export const getAllStakeholders = () => {
  return getStakeholders(
    {},
    { page: 1, perPage: 1000 },
    { field: StakeholderSortBy.DISPLAY_NAME }
  );
};

export const getAllStakeholderGroups = () => {
  return getStakeholderGroups(
    {},
    { page: 1, perPage: 1000 },
    { field: StakeholderGroupSortBy.NAME }
  );
};

export const getAllJobFunctions = () => {
  return getJobFunctions(
    {},
    { page: 1, perPage: 1000 },
    { field: JobFunctionSortBy.ROLE }
  );
};

export const getAllTagTypes = (
  sortBy: TagTypeSortByQuery = { field: TagTypeSortBy.NAME }
) => {
  return getTagTypes({}, { page: 1, perPage: 1000 }, sortBy);
};

export const getAllApplications = () => {
  return getApplications(
    {},
    { page: 1, perPage: 1000 },
    { field: ApplicationSortBy.NAME }
  );
};

export const getAllApplicationDependencies = (filters: {
  from?: string[];
  to?: string[];
}) => {
  return getApplicationDependencies(filters, { page: 1, perPage: 1000 });
};

//

export const stakeholderPageMapper = (
  page: StakeholderPage
): PageRepresentation<Stakeholder> => ({
  meta: { count: page.total_count },
  data: page._embedded.stakeholder,
});

export const stakeholderGroupPageMapper = (
  page: StakeholderGroupPage
): PageRepresentation<StakeholderGroup> => ({
  meta: { count: page.total_count },
  data: page._embedded["stakeholder-group"],
});

export const bussinessServicePageMapper = (
  page: BusinessServicePage
): PageRepresentation<BusinessService> => ({
  meta: { count: page.total_count },
  data: page._embedded["business-service"],
});

export const jobFunctionPageMapper = (
  page: JobFunctionPage
): PageRepresentation<JobFunction> => ({
  meta: { count: page.total_count },
  data: page._embedded["job-function"],
});

export const tagTypePageMapper = (
  page: TagTypePage
): PageRepresentation<TagType> => ({
  meta: { count: page.total_count },
  data: page._embedded["tag-type"],
});

//

export const applicationPageMapper = (
  page: ApplicationPage
): PageRepresentation<Application> => ({
  meta: { count: page.total_count },
  data: page._embedded.application,
});

export const applicationDependencyPageMapper = (
  page: ApplicationDependencyPage
): PageRepresentation<ApplicationDependency> => ({
  meta: { count: page.total_count },
  data: page._embedded["applications-dependency"],
});

export const applicationImportSummaryPageMapper = (
  page: ApplicationImportSummaryPage
): PageRepresentation<ApplicationImportSummary> => ({
  meta: { count: page.total_count },
  data: page._embedded["import-summary"],
});

export const applicationImportPageMapper = (
  page: ApplicationImportPage
): PageRepresentation<ApplicationImport> => ({
  meta: { count: page.total_count },
  data: page._embedded["application-import"],
});

//

export const IdentityPageMapper = (
  page: IdentityPage
): PageRepresentation<Identity> => ({
  meta: { count: page.total_count },
  data: page._embedded["identity"],
});

//

export const fetchAllPages = <T, P>(
  fetchPage: (page: number) => AxiosPromise<P>,
  responseToItems: (responseData: P) => T[],
  responseToTotalCount: (responseData: P) => number,
  page: number = 1,
  initialItems: T[] = []
): Promise<T[]> => {
  return fetchPage(page).then(({ data }) => {
    const accumulator = [...initialItems, ...responseToItems(data)];
    if (accumulator.length < responseToTotalCount(data)) {
      return fetchAllPages(
        fetchPage,
        responseToItems,
        responseToTotalCount,
        page + 1,
        accumulator
      );
    } else {
      return accumulator;
    }
  });
};
