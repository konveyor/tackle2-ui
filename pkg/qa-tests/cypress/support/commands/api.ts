/// <reference types="cypress" />

import { buildHeadersWithHAL_JSON, buildHeadersWithJSON } from "./utils";

const sizeQueryParam = "size=1000";
const controlsBaseUrl = "/api/controls";
const applicationInventoryBaseUrl = "/api/application-inventory";
const pathfinderBaseUrl = "/api/pathfinder";

export type ResourceType =
  | "JobFunction"
  | "Stakeholder"
  | "StakeholderGroup"
  | "BusinessService"
  | "TagType"
  | "Tag"
  | "Application"
  | "ApplicationsDependency"
  | "ImportSummary"
  | "Assessment";

const allResourceTypes: ResourceType[] = [
  "JobFunction",
  "Stakeholder",
  "StakeholderGroup",
  "BusinessService",
  "TagType",
  "Tag",
  "Application",
  "ApplicationsDependency",
  "ImportSummary",
  // "Assessment",
];

type ResourceTypeList = {
  [key in ResourceType]: {
    baseUrl: string;
    _embeddedField: string;
  };
};

const resourceTypeList: ResourceTypeList = {
  JobFunction: {
    baseUrl: `${controlsBaseUrl}/job-function`,
    _embeddedField: "job-function",
  },
  Stakeholder: {
    baseUrl: `${controlsBaseUrl}/stakeholder`,
    _embeddedField: "stakeholder",
  },
  StakeholderGroup: {
    baseUrl: `${controlsBaseUrl}/stakeholder-group`,
    _embeddedField: "stakeholder-group",
  },
  BusinessService: {
    baseUrl: `${controlsBaseUrl}/business-service`,
    _embeddedField: "business-service",
  },
  TagType: {
    baseUrl: `${controlsBaseUrl}/tag-type`,
    _embeddedField: "tag-type",
  },
  Tag: {
    baseUrl: `${controlsBaseUrl}/tag`,
    _embeddedField: "tag",
  },
  Application: {
    baseUrl: `${applicationInventoryBaseUrl}/application`,
    _embeddedField: "application",
  },
  ApplicationsDependency: {
    baseUrl: `${applicationInventoryBaseUrl}/applications-dependency`,
    _embeddedField: "applications-dependency",
  },
  ImportSummary: {
    baseUrl: `${applicationInventoryBaseUrl}/import-summary`,
    _embeddedField: "import-summary",
  },
  Assessment: {
    baseUrl: `${pathfinderBaseUrl}/assessments`,
    _embeddedField: undefined,
  },
};

const create = (headers: any, url: string, payload: any) => {
  cy.request({
    method: "POST",
    headers: headers,
    body: payload,
    url: url,
  }).its("body");
};

const singleRequest = (
  headers: any,
  url: string,
  payload: any,
  method: "PUT" | "PATCH" | "DELETE"
) => {
  cy.request({
    method: method,
    headers: headers,
    body: payload,
    url: `${url}/${payload.id}`,
  }).its("body");
};

const deleteAll = (headers: any, url: string, embeddedField: string) => {
  cy.request({
    method: "GET",
    headers: headers,
    url: `${url}?${sizeQueryParam}`,
  })
    .then((response) =>
      embeddedField ? response.body._embedded[embeddedField] : response.body
    )
    .each((item: any) =>
      cy.request({
        method: "DELETE",
        headers: headers,
        url: `${url}/${item.id}`,
      })
    );
};

// Utils

const getHeadersFromResourceType = (
  resourceType: ResourceType,
  tokens: KcTokens
) => {
  let headers;
  switch (resourceType) {
    case "Assessment":
      headers = buildHeadersWithJSON(tokens);
      break;
    default:
      headers = buildHeadersWithHAL_JSON(tokens);
      break;
  }
  return headers;
};
// Commands

declare global {
  namespace Cypress {
    interface Chainable<Subject> {
      api_crud(
        tokens: KcTokens,
        resource: ResourceType,
        method: "POST" | "PUT" | "PATCH" | "DELETE",
        payload: any
      ): Chainable<any>;
      api_clean(tokens: KcTokens, resource?: ResourceType): void;
    }
  }
}

Cypress.Commands.add(
  "api_crud",
  (
    tokens: KcTokens,
    resource: ResourceType,
    method: "POST" | "PUT" | "PATCH" | "DELETE",
    payload: any
  ) => {
    const headers = getHeadersFromResourceType(resource, tokens);
    const type = resourceTypeList[resource];

    switch (method) {
      case "POST":
        create(headers, type.baseUrl, payload);
        break;
      case "PUT":
      case "PATCH":
      case "DELETE":
        singleRequest(headers, type.baseUrl, payload, method);
        break;
    }
  }
);

Cypress.Commands.add(
  "api_clean",
  (tokens: KcTokens, resource?: ResourceType) => {
    const headers = getHeadersFromResourceType(resource, tokens);

    if (resource) {
      const type = resourceTypeList[resource];
      deleteAll(headers, type.baseUrl, type._embeddedField);
    } else {
      allResourceTypes.forEach((e) => {
        const type = resourceTypeList[e];
        deleteAll(headers, type.baseUrl, type._embeddedField);
      });
    }
  }
);

export {};
