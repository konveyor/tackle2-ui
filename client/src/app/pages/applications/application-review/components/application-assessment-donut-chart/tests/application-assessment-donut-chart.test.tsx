import React from "react";

import {
  ChartData,
  ApplicationAssessmentDonutChart,
  getChartDataFromCategories,
} from "../application-assessment-donut-chart";
import { Assessment } from "@app/api/models";
import { render } from "@app/test-config/test-utils";

describe("AppTable", () => {
  const assessments = [];
  // const assessment: Assessment = {
  //   id: 209,
  //   applicationId: 1,
  //   status: "COMPLETE",
  //   stakeholders: [],
  //   stakeholderGroups: [],
  //   questionnaire: {
  //     section: [
  //       {
  //         id: 211,
  //         order: 1,
  //         title: "Category1",
  //         questions: [
  //           {
  //             id: 212,
  //             order: 1,
  //             question: "question1",
  //             options: [
  //               {
  //                 id: 214,
  //                 order: 1,
  //                 option: "option",
  //                 checked: true,
  //                 risk: "GREEN",
  //               },
  //               {
  //                 id: 215,
  //                 order: 2,
  //                 option: "option",
  //                 checked: false,
  //                 risk: "AMBER",
  //               },
  //               {
  //                 id: 216,
  //                 order: 3,
  //                 option: "option",
  //                 checked: false,
  //                 risk: "RED",
  //               },
  //               {
  //                 id: 213,
  //                 order: 0,
  //                 option: "option",
  //                 checked: false,
  //                 risk: "GREEN",
  //               },
  //             ],
  //             description: "",
  //           },
  //           {
  //             id: 219,
  //             order: 2,
  //             question: "question2",
  //             options: [
  //               {
  //                 id: 221,
  //                 order: 1,
  //                 option: "option",
  //                 checked: false,
  //                 risk: "GREEN",
  //               },
  //               {
  //                 id: 222,
  //                 order: 2,
  //                 option: "option",
  //                 checked: true,
  //                 risk: "AMBER",
  //               },
  //               {
  //                 id: 223,
  //                 order: 3,
  //                 option: "option",
  //                 checked: false,
  //                 risk: "RED",
  //               },
  //               {
  //                 id: 224,
  //                 order: 4,
  //                 option: "option",
  //                 checked: false,
  //                 risk: "UNKNOWN",
  //               },
  //             ],
  //             description: "",
  //           },
  //           {
  //             id: 226,
  //             order: 3,
  //             question: "question3",
  //             options: [
  //               {
  //                 id: 228,
  //                 order: 1,
  //                 option: "option",
  //                 checked: false,
  //                 risk: "GREEN",
  //               },
  //               {
  //                 id: 229,
  //                 order: 2,
  //                 option: "option",
  //                 checked: false,
  //                 risk: "AMBER",
  //               },
  //               {
  //                 id: 230,
  //                 order: 3,
  //                 option: "option",
  //                 checked: true,
  //                 risk: "RED",
  //               },
  //               {
  //                 id: 231,
  //                 order: 4,
  //                 option: "option",
  //                 checked: false,
  //                 risk: "UNKNOWN",
  //               },
  //             ],
  //             description: "",
  //           },
  //           {
  //             id: 234,
  //             order: 4,
  //             question: "question3",
  //             options: [
  //               {
  //                 id: 236,
  //                 order: 1,
  //                 option: "option",
  //                 checked: false,
  //                 risk: "GREEN",
  //               },
  //               {
  //                 id: 237,
  //                 order: 2,
  //                 option: "option",
  //                 checked: false,
  //                 risk: "AMBER",
  //               },
  //               {
  //                 id: 238,
  //                 order: 3,
  //                 option: "option",
  //                 checked: false,
  //                 risk: "RED",
  //               },
  //               {
  //                 id: 235,
  //                 order: 0,
  //                 option: "option",
  //                 checked: true,
  //                 risk: "UNKNOWN",
  //               },
  //             ],
  //             description: "",
  //           },
  //         ],
  //         comment: "",
  //       },
  //       {
  //         id: 262,
  //         order: 2,
  //         title: "Category2",
  //         questions: [
  //           {
  //             id: 263,
  //             order: 1,
  //             question: "question4",
  //             options: [
  //               {
  //                 id: 264,
  //                 order: 0,
  //                 option: "option",
  //                 checked: false,
  //                 risk: "GREEN",
  //               },
  //               {
  //                 id: 266,
  //                 order: 2,
  //                 option: "option",
  //                 checked: false,
  //                 risk: "AMBER",
  //               },
  //               {
  //                 id: 267,
  //                 order: 3,
  //                 option: "option",
  //                 checked: false,
  //                 risk: "RED",
  //               },
  //               {
  //                 id: 268,
  //                 order: 4,
  //                 option: "option",
  //                 checked: true,
  //                 risk: "UNKNOWN",
  //               },
  //             ],
  //             description: "",
  //           },
  //         ],
  //         comment: "",
  //       },
  //     ],
  //   },

  it.skip("Get chart data", () => {
    // const result = getChartDataFromCategories(
    //   assessment.questionnaire.categories
    // );
    // expect(result).toMatchObject({
    //   green: 1,
    //   amber: 1,
    //   red: 1,
    //   unknown: 2,
    // } as ChartData);
  });

  it.skip("Renders without crashing", () => {
    // const wrapper = render(
    //   <ApplicationAssessmentDonutChart assessment={assessment} />
    // );
    // expect(wrapper).toMatchSnapshot();
  });
});
