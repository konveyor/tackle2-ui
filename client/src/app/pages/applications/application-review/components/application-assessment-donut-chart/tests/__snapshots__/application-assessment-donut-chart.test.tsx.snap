// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTable Renders without crashing 1`] = `
<div
  style={
    Object {
      "height": "250px",
      "width": "380px",
    }
  }
>
  <ChartDonut
    ariaDesc="risk-donut-chart"
    colorScale={
      Array [
        "#68b240",
        "#f0ab0b",
        "#cb440d",
        "#b8bbbe",
      ]
    }
    constrainToVisibleArea={true}
    data={
      Array [
        Object {
          "x": "risks.low",
          "y": 1,
        },
        Object {
          "x": "risks.medium",
          "y": 1,
        },
        Object {
          "x": "risks.high",
          "y": 1,
        },
        Object {
          "x": "risks.unknown",
          "y": 2,
        },
      ]
    }
    innerRadius={50}
    labels={[Function]}
    legendComponent={
      <ChartLegend
        colorScale={
          Array [
            "#68b240",
            "#f0ab0b",
            "#cb440d",
            "#b8bbbe",
          ]
        }
        data={
          Array [
            Object {
              "name": "risks.low: 1",
            },
            Object {
              "name": "risks.medium: 1",
            },
            Object {
              "name": "risks.high: 1",
            },
            Object {
              "name": "risks.unknown: 2",
            },
          ]
        }
      />
    }
    legendOrientation="vertical"
    legendPosition="right"
    padding={
      Object {
        "bottom": 20,
        "left": 20,
        "right": 140,
        "top": 20,
      }
    }
    width={380}
  />
</div>
`;
